packages/dev/core/src/Inputs/scene.inputManager.ts

```diff
@@ -201,43 +224,37 @@ export class InputManager {
             }
         }
 
-        const isMeshPicked = pickResult && pickResult.hit && pickResult.pickedMesh ? true : false;
-        if (isMeshPicked) {
-            scene.setPointerOverMesh(pickResult!.pickedMesh, evt.pointerId, pickResult);
-
-            if (!scene.doNotHandleCursors && canvas && this._pointerOverMesh) {
-                const actionManager = this._pointerOverMesh._getActionManagerForTrigger();
-                if (actionManager && actionManager.hasPointerTriggers) {
-                    canvas.style.cursor = actionManager.hoverCursor || scene.hoverCursor;
-                }
-            }
-        } else {
-            scene.setPointerOverMesh(null, evt.pointerId, pickResult);
-        }
+        this._setCursorAndPointerOverMesh(pickResult, evt.pointerId, scene);
 
         for (const step of scene._pointerMoveStage) {
+            const isMeshPicked = pickResult && pickResult.hit && pickResult.pickedMesh ? true : false;
             pickResult = step.action(this._unTranslatedPointerX, this._unTranslatedPointerY, pickResult, isMeshPicked, canvas);
         }
 
-        if (pickResult) {
-            const type = evt.inputIndex >= PointerInput.MouseWheelX && evt.inputIndex <= PointerInput.MouseWheelZ ? PointerEventTypes.POINTERWHEEL : PointerEventTypes.POINTERMOVE;
+        const type = evt.inputIndex >= PointerInput.MouseWheelX && evt.inputIndex <= PointerInput.MouseWheelZ ? PointerEventTypes.POINTERWHEEL : PointerEventTypes.POINTERMOVE;
 
-            if (scene.onPointerMove) {
-                scene.onPointerMove(evt, pickResult, type);
-            }
+        if (scene.onPointerMove && pickResult) {
+            scene.onPointerMove(evt, pickResult, type);
+        }
 
-            if (scene.onPointerObservable.hasObservers()) {
-                const pi = new PointerInfo(type, evt, pickResult);
-                this._setRayOnPointerInfo(pi);
-                scene.onPointerObservable.notifyObservers(pi, type);
-            }
+        // Always generate a PointerInfo, event if we don't have a pickResult, save pointerInfo to use with getter for meshUnderPointer
+        const pi = pickResult ? new PointerInfo(type, evt, pickResult) : new PointerInfo(type, evt, null, this);
+        if (!pickResult) {
+            this._movePointerInfo = pi;
+        } else {
+            this._setRayOnPointerInfo(pi);
+        }

```

Comment:
confusing logic, try this:
```suggestion
        let pointerInfo: PointerInfo;
        if (pickResult) {
            pointerInfo = new PointerInfo(type, evt, pickResult);
            this._setRayOnPointerInfo(pointerInfo);
        } else {
            pointerInfo = new PointerInfo(type, evt, null, this);
            this._movePointerInfo = pi;
        }
```